spec/models/registration_spec.rb

```diff
@@ -1,24 +1,70 @@
-#!/bin/env ruby
-# encoding: utf-8
 require 'spec_helper'
 
-describe 'Registration' do
-  describe 'validations' do
+describe Registration do
+  subject { create(:registration) }
+
+  describe 'Validation' do
+
     it 'has a valid factory' do
       expect(build(:registration)).to be_valid
     end
 
+    it { should validate_presence_of(:user) }
+
+    it 'validates uniqueness of user in scope of conference' do
+      expect(build(:registration, conference: subject.conference, user: subject.user)).not_to be_valid
+    end
+
     describe 'registration_limit_not_exceed' do
-      it 'is not valid when limit exceeded' do
-        conference = build(:conference)
-        conference.registration_limit = 1
-        registration1 = build(:registration, conference: conference)
-        registration1.save
-        registration2 = build(:registration, conference: conference)
-        registration2.save
-        expect(conference.registrations.size).to be 1
-        expect(registration2.valid?).to be false
-        expect(registration2.errors.full_messages).to eq(['Registration limit exceeded'])
+      context 'registration_limit has exceeded' do
+        before do
+          registration = create(:registration)
+          registration.conference.update_attributes(registration_limit: 1)
+          @second_registration = build(:registration, conference: registration.conference)
+        end
+
+        it 'is not valid factory' do
+          expect(@second_registration.valid?).to be false
+          expect(@second_registration.errors.full_messages).to eq(['Registration limit exceeded'])
+        end
+      end
+    end
+  end
+
+  describe 'Associations' do
+    it { should belong_to(:user) }
+    it { should belong_to(:conference) }
+    it { should belong_to(:dietary_choice) }
+    it { should have_and_belong_to_many(:social_events) }
+    it { should have_and_belong_to_many(:events) }
+    it { should have_and_belong_to_many(:qanswers) }
+    it { should have_and_belong_to_many(:vchoices) }
+    it { should have_many(:events_registrations) }
+    it { should have_many(:workshops) }
+  end
+
+  describe 'callbacks' do
+    after { subject.run_callbacks(:create) }
+
+    it 'sets week of registration' do
```

Comment:
Callbacks are private methods (Testing them breaks encapsulation). Also, `subscribe_to_conference` and `send_registration_mail` are just delegating responsibility to  `Subscription` and `Mailbot` classes. So, `Subscription` and `Mailbot` classes should make sure that it does what is was asked to (ie we need to test it those classes).

Do you want me to test them regardless?